### PR TITLE
CCM-6405 adding initial repo sync script and workflow

### PR DESCRIPTION
**Pull request by aidenvaines-bjss**
## Description

- Adding a script to compare the repository template with the current repo and bring in changes.
- Adding github actions workflow to run and commit proposed changes

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by jamesthompson26-nhs — DISMISSED** [review]
Looks good to me. Nice work.

**Review by sidnhs — APPROVED** [review]

File: .github/workflows/scheduled-repository-template-sync.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      uses: actions/checkout@v3
+      with:
+        repository: nhsdigital/nhs-notify-repository-template
+        path: nhs-notify-repository-template
+
+    - name: Run syncronisation script
+      run: |
+        /scripts/githooks/sync-template-repo.sh
+
+    - name: Create Pull Request
```
Comment (m-houston) [inline .github/workflows/scheduled-repository-template-sync.yaml:31]: What happens if there is no drift?

File: scripts/githooks/sync-template-repo.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  fi
+
+  target_path="../$relative_path"
+  mkdir -p "$(dirname "$target_path")"
+
+  # Copy the file to the root directory if it doesn't exist or is different
+  if [ ! -f "$target_path" ] && [ "$changes_only" == false ]; then
+    echo "Copying $relative_path to the repository"
+    FILES_ADDED+=("${relative_path}")
+    # cp "$file" "$target_path"
```
Comment (m-houston) [inline scripts/githooks/sync-template-repo.sh:82]: Are you intending this to be a dry run? Should this be uncommented before merge?

**Review by m-houston — APPROVED** [review]

File: .github/workflows/scheduled-repository-template-sync.yaml (hunk 1 above)
Comment (aidenvaines-bjss) [reply .github/workflows/scheduled-repository-template-sync.yaml:31]: according to the behaviours on the action

> If there are no changes (i.e. no diff exists with the checked-out base branch), no pull request will be created and the action exits silently.

https://github.com/peter-evans/create-pull-request/tree/v7.0.1/?tab=readme-ov-file#action-behaviour

File: scripts/githooks/sync-template-repo.sh (hunk 2 above)
Comment (aidenvaines-bjss) [reply scripts/githooks/sync-template-repo.sh]: good spot, fix pushed

**Review by m-houston — APPROVED** [review]

**Pull request merged**